### PR TITLE
In-App Purchase

### PR DESCRIPTION
**Pull request by supperspidey**
#### Ticket

N/A

#### Pull Request Checklist

- [x] Resolved any merge conflicts
- [x] No build errors or warnings are introduced
- [x] New files are written in Swift
- [x] New classes contain license headers
- [x] New classes have Documentation
- [x] New public methods have Documentation

#### Issue Summary

This PR contains IAP work.

#### Implementation Summary

- Included file size into the error object.

#### Reviewer Tips

N/A

#### How to Test

N/A

### CODE REVIEW

File: VimeoUpload/Upload/Operations/Async/ExportSessionExportCreateVideoOperation.swift [hunk 1]
```diff
@@ -162,9 +162,23 @@ open class ExportSessionExportCreateVideoOperation: ConcurrentOperation
                     return
                 }
                 
-                if let error = operation.error
+                if var error = operation.error
```
Comment (nicolelehrer) [inline VimeoUpload/Upload/Operations/Async/ExportSessionExportCreateVideoOperation.swift]: Can we revert this back to let?

File: VimeoUpload/Upload/Operations/Async/ExportSessionExportCreateVideoOperation.swift [hunk 2]
```diff
@@ -162,9 +162,23 @@ open class ExportSessionExportCreateVideoOperation: ConcurrentOperation
...
                 {
-                    strongSelf.error = error
+                    var userInfo = error.userInfo
+                    
+
```
Comment (nicolelehrer) [inline VimeoUpload/Upload/Operations/Async/ExportSessionExportCreateVideoOperation.swift]: Nit: extra spaces

File: VimeoUpload/Upload/Operations/Async/ExportSessionExportCreateVideoOperation.swift [hunk 3]
```diff
@@ -162,9 +162,23 @@ open class ExportSessionExportCreateVideoOperation: ConcurrentOperation
...
+                    
+                    let asset = AVURLAsset(url: url)
+                    do
+                    {
+                        let fileSize = try asset.fileSize()
+                        userInfo.append([UploadErrorKey.FileSize.rawValue : fileSize])
+                    }
+                    catch
+                    {
+
```
Comment (nicolelehrer) [inline VimeoUpload/Upload/Operations/Async/ExportSessionExportCreateVideoOperation.swift]: If using do-catch, we should handle the error here if calculating the file size fails. If there's no need to handle the error (e.g. we just don't append file size) can we then do 

```swift
if let fileSize = try? asset.fileSize()
{
     userInfo.append([UploadErrorKey.FileSize.rawValue : fileSize])
} 
```

**Comment by cameo-github** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          6 Warnings
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/ab339f061bff0c614532103a6e2f5cadb77b5377/VimeoUpload/Upload/Descriptor System/Old Upload/OldUploadDescriptor.swift#L50">VimeoUpload/Upload/Descriptor System/Old Upload/OldUploadDescriptor.swift#L50</a></strong>: string interpolation produces a debug description for an optional value; did you mean to make this explicit?  <br /><code>
            print("\(self.currentRequest.rawValue) \(self.identifier)")
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/ab339f061bff0c614532103a6e2f5cadb77b5377/VimeoUpload/UIKit/DescriptorKVObserver.swift#L31">VimeoUpload/UIKit/DescriptorKVObserver.swift#L31</a></strong>: extraneous duplicate parameter name; ‘descriptorType’ already has an argument label  <br /><code>
    func descriptorStateDidChange(descriptorType descriptorType: VideoDescriptorType, descriptorState: DescriptorState, isCancelled: Bool, descriptorError: NSError?)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/ab339f061bff0c614532103a6e2f5cadb77b5377/VimeoUpload/UIKit/DescriptorKVObserver.swift#L32">VimeoUpload/UIKit/DescriptorKVObserver.swift#L32</a></strong>: extraneous duplicate parameter name; ‘descriptorType’ already has an argument label  <br /><code>
    func descriptorProgressDidChange(descriptorType descriptorType: VideoDescriptorType, progress: Double)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/ab339f061bff0c614532103a6e2f5cadb77b5377/VimeoUpload/Descriptor System/DescriptorManager.swift#L232">VimeoUpload/Descriptor System/DescriptorManager.swift#L232</a></strong>: conditional downcast from ‘Error?’ to ‘NSError’ is a bridging conversion; did you mean to use ‘as’?  <br /><code>
                let isConnectionError = ((task.error as? NSError)?.isConnectionError() == true || (error as? NSError)?.isConnectionError() == true)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/ab339f061bff0c614532103a6e2f5cadb77b5377/VimeoUpload/Upload/Extensions/AVAsset+Extensions.swift#L32">VimeoUpload/Upload/Extensions/AVAsset+Extensions.swift#L32</a></strong>: extraneous duplicate parameter name; ‘completion’ already has an argument label  <br /><code>
    func approximateFileSize(completion completion: @escaping FloatBlock)
</code></td>
    </tr>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false"><strong><a href="https://github.com/vimeo/VimeoUpload/blob/ab339f061bff0c614532103a6e2f5cadb77b5377/VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift#L27">VimeoUpload/Upload/Descriptor System/Cameo/CAMUploadDescriptor.swift#L27</a></strong>: string interpolation produces a debug description for an optional value; did you mean to make this explicit?  <br /><code>
            print("\(self.currentRequest.rawValue) \(self.identifier)")
</code></td>
    </tr>
  </tbody>
</table>
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Message">
          2 Messages
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.885 (0.887) seconds</td>
    </tr>
    <tr>
      <td>:book:</td>
      <td data-sticky="false">Executed 2 tests, with 0 failures (0 unexpected) in 0.344 (0.348) seconds</td>
    </tr>
  </tbody>
</table>

See build details on [CircleCI](https://circleci.com/gh/vimeo/VimeoUpload/243)

<p align="right" data-meta="generated_by_danger">
  Generated by :no_entry_sign: <a href="http://danger.systems/">danger</a>
</p>

**Pull request merged**